starter-code/src/clue.js

```diff
@@ -1,94 +1,199 @@
+var pistol = {weight: 20};
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+var  DinningRoom = {};
+var  Conservatory = {};
+var  Kitchen = {};
+var  Study = {};
+var  Library = {};
+var  BilliardRoom = {};
+var  Lounge = {};
+var  Ballroom = {};
+var  Hall = {};
+var  Spa = {};
+var  LivingRoom = {};
+var  Observatory = {};
+var  Theater = {};
+var  GuestHouse = {};
+var  Patio = {};
+
 
 // Characters Collection
 var charactersArray = [];
 
+charactersArray.push(mrGreen);
+charactersArray.push(drOrchid);
+charactersArray.push(profPlum);
+charactersArray.push(missScarlet);
+charactersArray.push(mrsPeacock);
+charactersArray.push(mrMustard);
+
 // Rooms' Collection
 var roomsArray = [];
 
+roomsArray.push(DinningRoom);
+roomsArray.push(Conservatory);
+roomsArray.push(Kitchen);
+roomsArray.push(Study);
+roomsArray.push(Library);
+roomsArray.push(BilliardRoom);
+roomsArray.push(Lounge);
+roomsArray.push(Ballroom);
+roomsArray.push(Hall);
+roomsArray.push(Spa);
+roomsArray.push(LivingRoom);
+roomsArray.push(Observatory);
+roomsArray.push(Theater);
+roomsArray.push(GuestHouse);
+roomsArray.push(Patio);
+
+
+
+
 // Weapons Collection
 var weaponsArray = [];
+
+weaponsArray.push(rope);
+weaponsArray.push(knife);
+weaponsArray.push(candlestick);
+weaponsArray.push(dumbbell);
+weaponsArray.push(poison);
+weaponsArray.push(axe);
+weaponsArray.push(bat);
+weaponsArray.push(trophy);
+weaponsArray.push(pistol);
+
+
+// Characters
+/*Create a method randomSelector to randomly select one element 
+from a card stack. The method should receive 
+an array as an argument, and return randomly o
+ne of the elements of the array.*/
+
+function randomSelector(array) {
+
+	if (array.lenght === 0){
+		return false;
+	}
+
+	let rand = array[Math.floor(Math.random()*array.length)];
+
+
+	return rand;
+
+
+}
+
+/*We need to create a pickMistery method that 
+will call randomSelector for each card stack, 
+and return an array with the 3 picked cards, a 
+character, a weapon and a room. Our mystery 
+should be stored on a misteryEnvelope variable.*/
+
+
+function pickMistery(){
+
+    var misteryEnvelope = [];
+
+	let character = randomSelector(charactersArray);
+	let weapon = randomSelector(weaponsArray);
+	let room = randomSelector(roomsArray);
+
+
+	misteryEnvelope.push(character);
+	misteryEnvelope.push(weapon);
+	misteryEnvelope.push(room);
+
+	return misteryEnvelope;
+}
+
+/*Finally, we need to reveal the mystery. 
+Create a revealMistery method, that will receive 
+our misteryEnvelope array as an argument and 
+return the revealed mystery like this: 
+<FIRST NAME> <LAST NAME> killed Mr.Boddy using the 
+<WEAPON> in the <PLACE>!!!!*/
+
+function revealMistery(misteryEnvelope){
+
+
+var hello = "";
+
+var hello = (misteryEnvelope[0].first_name + " " + 
```

Comment:
Every variable that you create takes part of the memory so try not to create variables when there's no need. You can just:
```
return `${misteryEnvelope[0].first_name} ${misteryEnvelope[0].last_name} killed Mr.Boddy using the ${misteryEnvelope[1].name} in the ${misteryEnvelope[2].roomNames}!!!!`
```